src/elements/datepicker/datepicker-toggle/datepicker-toggle.visual.spec.ts

```diff
@@ -0,0 +1,90 @@
+import { html } from 'lit';
+
+import {
+  describeEach,
+  describeViewports,
+  visualDiffDefault,
+  visualDiffFocus,
+  visualRegressionFixture,
+} from '../../core/testing/private.js';
+
+import './datepicker-toggle.js';
+
+import '../datepicker.js';
+import '../../form-field.js';
+import '../../calendar.js';
+import type { SbbDatepickerToggleElement } from './datepicker-toggle.js';
+
+describe(`sbb-datepicker-toggle`, () => {
+  describeViewports({ viewports: ['wide'] }, () => {
+    describe('render', () => {
+      for (const state of [visualDiffDefault, visualDiffFocus]) {
+        let root: HTMLElement;
+        beforeEach(async () => {
+          root = await visualRegressionFixture(
+            html`
+              <div style="height: 600px;">
+                <div style="display: flex; gap: 1rem;">
+                  <sbb-datepicker-toggle
+                    date-picker="datepicker"
+                    id="toggle"
+                  ></sbb-datepicker-toggle>
+                  <sbb-datepicker
+                    id="datepicker"
+                    input="datepicker-input"
+                    now="2023-01-12T00:00:00Z"
+                  ></sbb-datepicker>
+                  <input id="datepicker-input" />
+                </div>
+              </div>
+            `,
+            {
+              padding: '4rem',
+            },
+          );
+        });
+
+        it(
```

Comment:
```suggestion
      for (const state of [visualDiffDefault, visualDiffFocus]) {
        it(
```

You don't want the beforeEach inside the loop because it would run twice for each case (you would have the template doubled)